pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -248,6 +254,17 @@ def __init__(
         self.__init_triggers(every_n_train_steps, every_n_epochs, train_time_interval, period)
         self.__validate_init_configuration()
 
+    @property
+    def state_key(self) -> str:
+        return self._generate_state_key(
+            monitor=self.monitor,
+            mode=self.mode,
+            every_n_train_steps=self._every_n_train_steps,
+            every_n_epochs=self._every_n_epochs,
+            train_time_interval=self._train_time_interval,
+            save_on_train_epoch_end=self._save_on_train_epoch_end,
+        )
+
```

Comment:
since it'll come up: how do we handle deprecated arguments being used to generate the state key?

Reply:
We are not handling it in this PR. This is planned for 3/3 or 4/4 if necessary. 
I think the only way we will be able to handle this will be to provide an upgrade function whenever we introduce name changes etc. that upgrades the keys in the checkpoint dict before we fully load the state into the objects.

Reply:
This was the plan for 3/3, that PR exists but is completely outdated now.